pkg/runtime/runtime.go

```diff
@@ -2826,6 +3084,16 @@ func (a *DaprRuntime) authSecretStoreOrDefault(comp componentsV1alpha1.Component
 	return comp.SecretStore
 }
 
+func (a *DaprRuntime) authSecretStoreOrDefaultHTTPEndpoint(endpoint httpEndpointV1alpha1.HTTPEndpoint) string {
```

Comment:
If you define an interface that includes the method `GetSecretStore()` (which just returns the property `SecretStore`) on both `httpEndpointV1alpha1.HTTPEndpoint` and `componentsV1alpha1.Component`, you can avoid implementing this method twice. Please do so to avoid code repetition and make this simpler to manage in the future.

Reply:
went ahead and made the func accept an interface and have a switch checking types. Please let me know if you'd rather the other approach.